### PR TITLE
chore(deps): upgrade k8s version and client-go

### PR DESCRIPTION
**Pull request by blakepettersson**
This is a continuation of #13965, but with the merge conflicts fixed.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

**Comment by blakepettersson** [comment]
@crenshaw-dev this branch is looking good now 😄

File: go.mod [hunk 1]
```diff
@@ -274,18 +275,19 @@ require (
...
+	k8s.io/gengo v0.0.0-20220902162205-c0856e24416d // indirect
+	k8s.io/kube-aggregator v0.26.4 // indirect
+	k8s.io/kubernetes v1.26.4 // indirect
+	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
+	sigs.k8s.io/kustomize/api v0.12.1 // indirect
+	sigs.k8s.io/kustomize/kyaml v0.13.9 // indirect
 )
 
 replace (
+	github.com/argoproj/gitops-engine => github.com/fengshunli/gitops-engine v0.7.1-0.20230922195635-cbe8d1c14245
```
Comment (crenshaw-dev) [inline go.mod:290]: @blakepettersson can you update this now that I've merged the gitops-engine PR? https://github.com/argoproj/gitops-engine/pull/530

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Comment by blakepettersson** [comment]
@crenshaw-dev done 😄

**Review by crenshaw-dev — APPROVED** [review]
Heck yeah

**Pull request merged**